seaborn/utils.py

```diff
@@ -414,3 +418,37 @@ def axes_ticklabels_overlap(ax):
     """
     return (axis_ticklabels_overlap(ax.get_xticklabels()),
             axis_ticklabels_overlap(ax.get_yticklabels()))
+
+
+def categorical_order(values, order=None):
+    """Return a list of unique data values.
+
+    Determine an ordered list of levels in ``values``.
+
+    Parameters
+    ----------
+    values : list, array, Categorical, or Series
+        Vector of "categorical" values
+    order : list-like, optional
+        Desired order of category levels to override the order determined
+        from the ``values`` object.
+
+    Returns
+    -------
+    order : list
+        Ordered list of category levels
+
+    """
+    if order is None:
+        if hasattr(values, "categories"):
```

Comment:
Just fore reference: this only supports pandas >=0.15, in former versions `Categorical.levels` have to be used...